frontend/actions/index.js

```diff
@@ -31,3 +49,68 @@ export const fetchPointsProducts = () => (dispatch, getState) => {
     });
 };
 
+/**
+ * Fetches Smile Customer from smile
+ * @returns {Function}
+ */
+export const fetchSmileCustomer = () => (dispatch, getState) => {
+  const smileCustomerState = getSmileCustomerState(getState()) || {};
+
+  if (smileCustomerState.isFetching || smileCustomerState.pointsProducts) {
+    return;
+  }
+
+  dispatch(requestSmileCustomer());
+  new PipelineRequest(GET_SMILE_CUSTOMER)
+    .dispatch()
+    .then((response) => {
+      const { customer } = response || {};
+      dispatch(receiveSmileCustomer(customer));
+    })
+    .catch((err) => {
+      logger.error(err);
+      dispatch(errorSmileCustomer());
+    });
+};
+
+/**
+ * Fetches Smile Your Rewards
+ * @return {Function}
+ */
+export const fetchSmileYourRewards = () => (dispatch) => {
+  dispatch(requestSmileYourRewards());
```

Comment:
no isFetching check here?

Reply:
This already looks done to me.

Reply:
There looks to be a fetching check. I don't know why it doesn't appear in this view?